lua/wikis/commons/MvpTable.lua

```diff
@@ -154,56 +156,57 @@ end
 
 ---Builds the main header of the MvpTable
 ---@param args mvpTableParsedArgs
----@return Html?
+---@return Widget?
 function MvpTable._mainHeader(args)
 	if String.isEmpty(args.title) then
-		return nil
+		return
 	end
 
-	local colspan = 2 + (args.points and 1 or 0)
-
-	return mw.html.create('tr')
-		:tag('th'):wikitext(args.title):attr('colspan', colspan):done():done()
+	return HtmlWidgets.Tr{
+		children = HtmlWidgets.Th{
+			attributes = {colspan = 2 + (args.points and 1 or 0)},
+			children = args.title
+		}
+	}
 end
 
 ---Builds the sub header of the MvpTable
 ---@param args mvpTableParsedArgs
----@return Html
+---@return Widget
 function MvpTable._subHeader(args)
-	local header = mw.html.create('tr')
-		:tag('th'):wikitext('Player'):done()
-		:tag('th'):wikitext('#MVPs'):done()
-
-	if args.points then
-		header:tag('th'):wikitext('Points'):done()
-	end
-
-	return header:done()
+	return HtmlWidgets.Tr{
+		children = Array.map(
+			{'Player', '#MVPs', args.points and 'Points' or nil},
+			function (element) return HtmlWidgets.Th{children = element} end
```

Comment:
oos, but would it be a good idea to add a function to widgetUtil that would map elements as children to a widget
i think we might be able to simplify a few places with it

```lua
---@param widget Widget
---@param elements ((Widget|string|number|Html)[]|Widget|string|number|Html?)[]?
---@return Widget[]?
function WidgetUtil.mapToWidget(widget, elements)
	if Logic.isDeepEmpty(elements) then return end
	return Array.map(elements, function(element)
		return widget{children = element}
	end)
end
```

usage here would be
```lua
	return HtmlWidgets.Tr{
		children = WidgetUtil.mapToWidget(HtmlWidgets.Th, {'Player', '#MVPs', args.points and 'Points' or nil})
	}
```

(untested, just an idea)